### PR TITLE
Bump maven-model-helper to 16

### LINKED ISSUE #12105: Maven Model Merger is messing with properties order in Codestarts
**Description**
properties are set in a specific order in the templates:
```xml
<properties>
        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
        <maven.compiler.source>{java.version}</maven.compiler.source>
        <maven.compiler.target>{java.version}</maven.compiler.target>
        <maven.compiler.parameters>true</maven.compiler.parameters>

        <quarkus.platform.group-id>{quarkus.platform.group-id}</quarkus.platform.group-id>
        <quarkus.platform.artifact-id>{quarkus.platform.artifact-id}</quarkus.platform.artifact-id>
        <quarkus.platform.version>{quarkus.platform.version}</quarkus.platform.version>
        <quarkus-plugin.version>{quarkus.maven-plugin.version}</quarkus-plugin.version>
        <compiler-plugin.version>{maven-compiler-plugin.version}</compiler-plugin.version>
        <surefire-plugin.version>{maven-surefire-plugin.version}</surefire-plugin.version>
    </properties>
```

then when merged, it ends up with something like:
```xml
<properties>
    <quarkus.platform.version>1.8.0.CR1</quarkus.platform.version>
    <scala-maven-plugin.version>4.1.1</scala-maven-plugin.version>
    <quarkus.platform.artifact-id>quarkus-bom</quarkus.platform.artifact-id>
    <compiler-plugin.version>3.8.1</compiler-plugin.version>
    <surefire-plugin.version>2.22.1</surefire-plugin.version>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
    <maven.compiler.parameters>true</maven.compiler.parameters>
    <maven.compiler.source>11</maven.compiler.source>
    <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
    <maven.compiler.target>11</maven.compiler.target>
    <scala.version>2.12.8</scala.version>
    <quarkus-plugin.version>1.8.0.CR1</quarkus-plugin.version>
    <quarkus.platform.group-id>io.quarkus</quarkus.platform.group-id>
  </properties>
```

I think it should just append the properties when merging (not reordering).

CC @gastaldi this is using the smart model merger.

### PR DESCRIPTION
**Pull request by gastaldi**
Fixes #12105

### CODE REVIEW

**Review by ia3andy — CHANGES_REQUESTED** [review]
@gastaldi you could add some properties there: https://github.com/gastaldi/quarkus/blob/bump_maven_merger/independent-projects/tools/codestarts/src/test/resources/codestarts/core/buildtool-maven/base/pom.tpl.qute.xml

then the https://github.com/gastaldi/quarkus/blob/bump_maven_merger/independent-projects/tools/codestarts/src/test/resources/expected-pom-maven-merge.xml

will need to be changed accordingly and you can make sure of the order when merged :)

**Comment by ia3andy** [comment]
Thanks @gastaldi I believe this is really important :)

**Review by ia3andy — APPROVED** [review]
LGTM

File: independent-projects/tools/codestarts/src/test/resources/expected-pom-maven-merge.xml [hunk 1]
```diff
@@ -6,8 +6,11 @@
   <artifactId>quarkus-project</artifactId>
   <version>1.2.3</version>
   <properties>
-    <prop2>prop-2-namespaced</prop2>
+    <abc>123</abc>
+    <fruit>pineapple</fruit>
     <prop1>proppp</prop1>
+    <prop2>prop-2-namespaced</prop2>
+    <zebra>starts-with-z</zebra>
```
Comment (ia3andy) [inline independent-projects/tools/codestarts/src/test/resources/expected-pom-maven-merge.xml:13]: @gastaldi why is zebra at the end?

Comment (ia3andy) [reply independent-projects/tools/codestarts/src/test/resources/expected-pom-maven-merge.xml:13]: It's not supposed to sort by name but by definition order right?

**Review by ia3andy — CHANGES_REQUESTED** [review]

**Comment by gastaldi** [comment]
@ia3andy Because it's impossible to preserve the order using the Maven Model API (they are read as Properties - which does not preserve the order of the keys), the model-merger must sort by name to be able to at least predict what the results are.

**Comment by ia3andy** [comment]
Well at least it will keep together 😅:
```xml
<maven.compiler.source>{java.version}</maven.compiler.source>
<maven.compiler.target>{java.version}</maven.compiler.target>
<maven.compiler.parameters>true</maven.compiler.parameters>
```

**Comment by gastaldi** [comment]
@ia3andy yeah, but the end result will be: 

```
<maven.compiler.parameters>true</maven.compiler.parameters>
<maven.compiler.source>{java.version}</maven.compiler.source>
<maven.compiler.target>{java.version}</maven.compiler.target>
```

:wink:

**Review by ia3andy — APPROVED** [review]
LGTM (best possible solution even if not perfect) Thanks!

**Pull request merged**